cluster-state-service/handler/httpclient/httpclient.go

```diff
@@ -47,10 +51,11 @@ func init() {
 
 // New returns an Blox httpClient that will insert custom HTTP UA header.
 func New() *http.Client {
-	transport := &http.Transport{}
+	transport := http.DefaultTransport
 
 	client := &http.Client{
 		Transport: &bloxRoundTripper{transport},
+		Timeout: httpClientTimeout,
```

Comment:
This is a hard timeout. The timeouts in `http.DefaultTransport` are more granular. It sets the keep-alive idle timeout to 90s and the connection timeout to 30s. This means that together the individual timeouts in `http.DefaultTransport` can actually exceed 30 seconds. By setting it on the `http.Client`, I am ensuring that the request should timeout if it takes more than 30s from the start, regardless of how those 30s are consumed.